src/main/java/net/countercraft/movecraft/combat/movecraftcombat/listener/PistonListener.java

```diff
@@ -0,0 +1,69 @@
+package net.countercraft.movecraft.combat.movecraftcombat.listener;
+
+import net.countercraft.movecraft.combat.movecraftcombat.config.Config;
+import org.bukkit.Location;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockPistonRetractEvent;
+import org.bukkit.util.Vector;
+
+public class PistonListener implements Listener
+{
+    @EventHandler
+    public void onPistonRetract(BlockPistonRetractEvent e) {
+        if(!Config.ReImplementTranslocation)
+            return;
+
+        Location headLocation = e.getBlock().getLocation().add(blockFaceToVector(e.getDirection()));
+        Vector direction = blockFaceToVector(e.getDirection()).multiply(-1D);
+        Location moveLocation = e.getBlock().getLocation().add(direction);
```

Comment:
You can use e.getBlock().getRelative(e.getDirection()) to simplify the head location.

Similarly, you can use e.getDirection().getOppositeFace() to get the moveLocation.  You'll still need to add 0.5 to get the correct center of the block location.